server/neptune/workflows/internal/pr/revision/processor.go

```diff
@@ -60,7 +60,11 @@ func (p *Processor) Process(ctx workflow.Context, prRevision Revision) {
 			}
 		}
 	}
-	p.PolicyHandler.Process(ctx, failedPolicies)
+	if len(failedPolicies) > 0 {
+		workflow.Go(ctx, func(c workflow.Context) {
+			p.PolicyHandler.Handle(ctx, prRevision, failedPolicies)
+		})
+	}
 }
```

Comment:
Instead of looping here to get the failed policies, I would just pass in the whole list and do the partitioning in the handler itself. Feels unnecessary for it to be here if we have a whole struct thats handling the result.

Reply:
Personally I preferred it this way to keep the PolicyHandler from expecting []ValidationResults, felt cleaner to me to call this a policy handler and just pass the relevant policies to it.

Reply:
> to keep the PolicyHandler from expecting []ValidationResults

Whats wrong with that?  I'm thinking about it from a pluggability standpoint here, if we need to add more logic that does something with policy results we're not limited to just the failing set.  So since there is only one thing using the failing set, just compute that failing set within the thing thats using it.